src/main/java/com/byclosure/webcat/stepdefinitions/helpers/SeleniumInteractions.java

```diff
@@ -0,0 +1,84 @@
+package com.byclosure.webcat.stepdefinitions.helpers;
+
+import com.byclosure.webcat.context.IContext;
+import org.openqa.selenium.*;
+import org.openqa.selenium.internal.selenesedriver.TakeScreenshot;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SeleniumInteractions {
+
+    private final WebDriver driver;
+    private final IContext context;
+
+    public SeleniumInteractions(WebDriver driver, IContext context) {
+        this.driver = driver;
+        this.context = context;
+    }
+
+    public void clickLink(String locator) {
+        WebElement linkElement = findLinkElement(locator);
+        linkElement.click();
+    }
+
+    public boolean searchForContent(String text) {
+        final WebElement body = driver.findElement(By.xpath("//*"));
+        return body.isDisplayed() && body.getText().contains(text);
+    }
+
+    public boolean searchForContent(String regex, String flags) {
+        final WebElement body = driver.findElement(By.xpath("//*"));
+        final String bodyContent = body.getText();
+        Matcher matcher = buildMatcherForContent(regex, flags, bodyContent);
+
+        return body.isDisplayed() && matcher.matches();
+    }
+
+    public void takeScreenshot() {
+        if(driver instanceof TakeScreenshot) {
+            final byte[] screenshot = ((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES);
+            context.addScreenshot(screenshot);
+        }
+    }
+
+    private WebElement findLinkElement(String locator) {
+        List<WebElement> elementsByText;
+        List<WebElement> elementsById;
+        List<WebElement> elementsByTitle;
+        List<WebElement> elementsByImageAlt;
+
+        if((elementsByText = driver.findElements(By.linkText(locator))).size() > 0) {
```

Comment:
don't assign and test in the same line of code..

If you get local variables final as it should be the code as-is would not work and it would force you to write better code.